File: java/postRequest.java
import java.io.IOException;
import java.io.InputStream;
import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.io.OutputStreamWriter;
import java.net.HttpURLConnection;
import java.net.URL;
import java.util.Date;
import java.text.DateFormat;
import java.text.SimpleDateFormat;
 
public class PostRequest {
  
     /*Note: This code example requires Base64Coder file which is 
available below in the Dependencies section.*/

  public static void main(String[] args) {
         
    String APIKey = "YOUR_API_KEY";
    String TeamworkURL = "https://YOUR_TEAMWORK_SITE_NAME.teamwork.com";
    String tasklistID = "TASK_LIST_ID";
    String taskName = "This is an example task created via a POST request in Java.";
    String action = "/todo_lists/" + tasklistID + "/todo_items.json";
         
    DateFormat dateFormat = new SimpleDateFormat("yyyyMMdd");
    Date date = new Date();
    String dueDate = dateFormat.format(date);
         
    String json = "{\"todo-item\":{\"content\":\""
                +taskName+"\",\"due-date\":\""+dueDate+"\"}}";
         
    try {
        URL url = new URL( TeamworkURL + action );
        HttpURLConnection connection = (HttpURLConnection) url.openConnection();
        connection.setRequestMethod("POST");
        connection.setDoOutput(true);
        connection.setRequestProperty("Content-Type", "application/json;charset=UTF-8");
             
        String userpassword = APIKey + ":xxx";
        String encodedAuthorization = Base64Coder.encodeString( userpassword );
             
        connection.setRequestProperty("Authorization", "BASIC "
                    + encodedAuthorization);
 
        OutputStreamWriter osw = new OutputStreamWriter(connection.getOutputStream());
        osw.write(json);
        osw.flush();
        osw.close();
 
        InputStream _is;
        if (connection.getResponseCode() != 201) {
        /* error  */
        errorStream = connection.getErrorStream();
        System.out.println( streamToString(errorStream) );
        }
 
        InputStream responseStream = connection.getInputStream();
        System.out.println( streamToString( responseStream) );
    } catch(Exception e) {
        System.out.println( "Error Received:" + e.getMessage() );
    }
  }
     
  public static String streamToString(InputStream in) throws IOException {
    StringBuilder out = new StringBuilder();
    BufferedReader br = new BufferedReader(new InputStreamReader(in));
    for(String line = br.readLine(); line != null; line = br.readLine()) 
    out.append(line);
    br.close();
    return out.toString();
  }
}

File: php/postRequest.php
<?php
$company = "YOUR_TEAMWORK_SITENAME";
$key = "YOUR_API_KEY";
$taskListId = 1;
$taskName = "This is an example task.";
$taskDate = time();
  
 
$arr = array(
    'todo-item' =>  array(
        'content'   =>  $taskName,
        'due-date'  =>  date('Ymd', $taskDate)
    )
);
 
$json = json_encode($arr);
 
$ch = curl_init();

$options = array(
    CURLOPT_URL => "https://{$company}.teamwork.com/todo_lists/{$taskListId}/todo_items.json",
    CURLOPT_RETURNTRANSFER => true,
    CURLOPT_POST => true,
    CURLOPT_POSTFIELDS => $json,
    CURLOPT_SSLVERSION => 6,
    CURLOPT_HTTPHEADER => array(
        "Authorization: BASIC ". base64_encode($key .":xxx"),
        "Content-type: application/json"
    )
);

curl_setopt_array($ch, $options);

$result = curl_exec($ch);
if (curl_error($ch)) {
    $error_msg = curl_error($ch);
}

if (isset($error_msg)) {
    // TODO - Handle cURL error
}

if ($result !== false) {
    $response = curl_getinfo($ch, CURLINFO_HTTP_CODE);  

    if ($response == 201) {
        // TODO - Handle cURL response
        $obj = json_decode($result);
        var_dump($obj);
    }
}

curl_close($ch);
